docs/Extensibility.md

```diff
@@ -0,0 +1,104 @@
+# Extensibility
+
+> Expect this pattern to evolve (as it has already done) or even disappear if [proper extensibility features](https://community.starknet.io/t/contract-extensibility-pattern/210/11?u=martriay) are implemented into Cairo.
+
+* [The extensibility problem](#the-extensibility-problem)
+* [The pattern ™️](#the-pattern)
+  * [Libraries](#libraries)
+  * [Contracts](#contracts)
+* [Presets](#presets)
+* [Emulating hooks](#emulating-hooks)
+
+## The extensibility problem
+
+Smart contract development is a critical task. As all software development, is error prone; but unlike most scenarios a bug can result in major losses for organizations as well as individuals. Therefore writing complex smart contracts is a delicate task.
+
+One of the best approaches to minimize introducing bugs is to reuse existing, battle-tested code, a.k.a. using libraries. But code reutilization in StarkNet’s smart contracts is not easy:
+
+- Cairo has no explicit smart contract extension mechanisms such as inheritance or composability
+- There’s no function overloading nor aliasing, making function selector collisions very likely – more so considering selectors do not take function arguments into account
+- Any `@external` function defined in an imported module will be automatically re-exposed by the importer (i.e. the smart contract)
+
+To overcome these problems, this library builds on the following guidelines™.
```

Comment:
Correct me if I'm wrong, but I believe this references the library of libraries here. I wonder if we should use a different name (e.g. collection of libraries, library of contracts). I think something akin to the suggestion adds clarity.
```suggestion
To overcome these problems, our StarkNet library builds on the following guidelines™.
```

Reply:
Good catch. "this repository"?

Reply:
"this repository" might be appropriate. There's also "this protocol"